### DISCUSSION TITLE
How to change folder structure on GitHub?

### DISCUSSION THREAD

**Discussion by NanaNiki** [community/community#57207] [New to GitHub] [answered]
Hi, 
I'm a freshie to coding and github, and i think i made a mistake at the beginning of building my portfolio project. I somehow created a remote repo but with undesired folder structure. 

Currently when you click on my repository, you can see one folder _my-app_, and than inside the _my-app_ folder there are folders and files. 

```
E-commerce
└── my-app
    └── components 
    └── data
    └── pages
    └── public
    └── ...more
```

This structure is identical with the structure of my local file. Also I have another repo of different project with the same local structure but it is accessing the content of the my-app folder directly on github, which is more convenient and desirable for deployment using gtihub. 

So my main question is how can i change the folder structure on github, without losing any git history, for the github repo to look like this:

```
E-commerce
    └── components 
    └── data
    └── pages
    └── public
    └── ...more
```

I've read about changing the root directory with git, but there is a high % of losing history? [https://stackoverflow.com/questions/11764003/change-the-root-directory-of-a-git-repository%20#%20/path/to/repo/but/i/want/the/target-dir/](url)
and also that is impossible? 
[https://stackoverflow.com/questions/35618507/git-sure-fire-way-to-move-rename-files-while-keeping-the-history?rq=3](url)

because from what i understand i might have make a repository in _E-commerce_ folder, not in _my-app_ folder that's why the botched structure. I mostly used git and github reating repos form vscode.

Kindly asking for your help! 
Sending love <3

**Comment by fsgreco** [comment, marked as answer]
The right way to do this should be using git `move` command; `git mv`  (it should preserve the history of the files - [git mv man page](https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-mv.html)).  

Are you using linux / osX ? in that case you can perform this simple operation with the terminal:
```shell
git mv ./my-app/* .
```
Then move manually the files `./my-app/.gitignore` and `./my-app/.eslintrc.json` to the root. 
(you can also do this via terminal: `mv ./my-app/{.gitignore,.eslintrc.json} .`  

And finally:
```shell
git rm ./my-app
git add -A
git commit -m "Folders moved out of my-app directory :)"
```

PS: you can try this without worries by cloning your repository on another folder and perform this operation, if the result is fine then push it online 👍🏽

**Reply by NanaNiki** [reply]
Sweet! <3! Thank you so much!
in my case the first part of syntax was: 
```
mv my-app/* .
mv my-app/.* .
```

because with _git_ in front it was giving me the error of:
_fatal: bad source, source=my-app/.*, destination=.*_

and than with the rest of Your code everything went smoothly !
🙏

**Reply by queenofcorgis** [reply]
Thank you so much @fsgreco for jumping in to help. Also kudos to @NanaNiki for coming back to mark the answer and wshare more details ❤️ .
